graph/patterns/navigation-property.md

```diff
@@ -0,0 +1,208 @@
+
+     -  /user/{userId}/manager  represents many-to-one relationship
+     -  /user/{userId}/messages represents one-to-many relationship
+
+Additionally, using the OData Expand query parameter, related entities can be nested into the primary entity so both can be retrieved in a single round trip.
+
+These relationships can be described in CSDL as follows:
+
+```xml
+<EntityType Name="user">
+  <NavigationProperty Name="manager" Type="user" ContainsTarget="false" >
+  <NavigationProperty Name="messages" Type="user" ContainsTarget="true" >
+</EntityType>
+```
+
+## Issues and Considerations
+-------------------------
+
+In the current Microsoft Graph implementation, there are some limitations on the use of navigation properties that cross between backend services. These limitations are being eliminated over time, but it will be necessary to ensure support for any particular scenario.  [Limitations of the current implementation](https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/354352/Cross-workload-navigations?anchor=supported-scenarios) are documented internally.
+ 
+Navigation properties defined within an entity are not returned by default when retreiving the representation of an entity unless explicity desired by a service.  The API can consumer can use the `expand` query parameterm, where supported, to retreive both the source and the target entity of the relationship in a single request.
+
+Implementing support for accessing the "$ref" of a navigation property allows a caller to return just the URL of related resource. e.g. `/user/23/manager/$ref`. This is useful when a client wishes to identify the related resource but doesn't need all of its properties.
+
+## When to Use this Pattern
+------------------------
+
+### "Many-to-one" relationships  
+
+The use of navigation properties is preferred over including an Id field to reference the related entity in a many-to-one relationship.  Id values require a client to make two round trips to retrieve the details of a related entity.  With a navigation property a client can retrieve a related entity in a single round trip. 
+ 
+Many-to-one relationships are always non-contained relationships as the lifetime of the target cannot depend on the source.
```

Comment:
Nit: Can we make "Many-to-one relationships are always non-contained relationships as the lifetime..." a little more precise for example "Many-to-one relationships are often modeled as non-contained relationships so that the lifetime..."?